org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMCommentFormatter.java

```diff
@@ -44,55 +50,78 @@ public void formatComment(DOMComment commentNode, XMLFormattingConstraints paren
 			leftWhitespaceOffset--;
 		}
 
+		// If preserve new lines
+		int preservedNewLines = getPreservedNewlines();
 		boolean addLineSeparator = formatterDocument.hasLineBreak(leftWhitespaceOffset, start);
-		if (addLineSeparator) {
-			// Indent in case of comment being at the start of line
-			replaceLeftSpacesWithIndentation(parentConstraints.getIndentLevel(), leftWhitespaceOffset, start, addLineSeparator, edits);
+		int indentLevel = parentConstraints.getIndentLevel();
+		int currentNewLineCount = XMLFormatterDocumentNew.getExistingNewLineCount(formatterDocument.getText(),
+				start,
+				formatterDocument.getLineDelimiter());
+		if (currentNewLineCount > preservedNewLines) {
+			replaceLeftSpacesWithIndentationWithMultiNewLines(indentLevel, 0, start,
+					preservedNewLines + 1, edits);
+			availableLineWidth = getMaxLineWidth() - getTabSize();
+		} else if (addLineSeparator) {
+			// remove spaces and indent
+			replaceLeftSpacesWithIndentation(parentConstraints.getIndentLevel(), leftWhitespaceOffset, start,
+					addLineSeparator, edits);
+			availableLineWidth = getMaxLineWidth() - getTabSize();
 		}
 		int spaceStart = -1;
 		int spaceEnd = -1;
-
-		if (isJoinCommentLines()) {
-			for (int i = commentNode.getStartContent(); i < commentNode.getEndContent(); i++) {
-				char c = text.charAt(i);
-				if (Character.isWhitespace(c)) {
-					// Whitespaces
-					if (spaceStart == -1) {
-						spaceStart = i;
-					} else {
-						spaceEnd = i;
-					}
+		availableLineWidth -= 4;
+
+		for (int i = commentNode.getStartContent(); i < commentNode.getEndContent(); i++) {
+			char c = text.charAt(i);
+			if (Character.isWhitespace(c)) {
+				// Whitespaces
+				if (spaceStart == -1) {
+					spaceStart = i;
+				} else {
+					spaceEnd = i;
+				}
+			} else {
+				spaceEnd = i;
+				// Ensure the edit is within the selected range
+				if (startRange != -1 && endRange != -1 && (startRange > spaceStart || endRange < spaceEnd)) {
```

Comment:
If I have:
```xml
<dress>


    <!--  as;ldkf j;lsa jdf;lsa jfd;lsa jfd;lakjs df;ljsfd;a jsdf
    ajs df
    asj dflk aslkjfas
    d kfaskdf jas jfdlkasj fd
    ka jdsfka
    sdf aksdfj aks
    jfdkajs fd
    kajs fdk  -->

<size></size>

</dress>
```

and `"xml.format.joinCommentLines": true,`, `"xml.format.maxLineWidth": 80,`

and I range format just the comment, it adjusts the indentation on `<size></size>`, even though that's not selected. I know this is a hard issue, but see if you can avoid adjusting the whitespace after the comment in this case.